books/tests/test_views.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+import pytest
+
+from rest_framework import status
+from rest_framework.utils import json
+from model_bakery import baker
+
+from books.models import Book
+from books.serializers import Bookserializer
+
+pytestmark = pytest.mark.django_db
+
+
+class TestViewBooks:
+    endpoint = '/api/books/'
+
+    def test_get_all_books(self, api_client, user_client):
+        baker.make(Book, _quantity=3)
+        user_client
+        response = api_client.get(
+            self.endpoint
+        )
+
+        assert response.status_code == status.HTTP_200_OK
+        assert isinstance(json.loads(response.content), list)
+        assert len(json.loads(response.content)) == 3
+
+    def test_post_book_success(self, api_client, user_client):
+        book = baker.prepare(Book)
+        body = {
+            "id": book.id,
+            "title": book.title,
+            "descriprion": book.descriprion,
+            "phone_number": book.phone_number,
+            "created_at": book.created_at,
+            "updated_at": book.updated_at,
+        }
+
+        user_client
+        response = api_client.post(
+            self.endpoint,
+            body,
+            format='json',
+        )
+
+        assert response.status_code == status.HTTP_201_CREATED
+        assert "id" in json.loads(response.content)
+
+    def test_post_book_invalid(self, api_client,         user_client):
+        body_invalid = {
+            "title": "title1231222223",
+            "descriprion": "lorem lorem lorem lorem lorem lorem lorem lorem lorem lorem",
+            "phone_number": "asdasdasd"
+        }
+
+        user_client
```

Comment:
What is the point of injecting user_client to force auth on the existing api_client, when a separate authenticated api client would do the work? api_client and auth_api_client should be two separate fixtures, independent from each other